### PR TITLE
[Next.js][Personalize] Switchover to Variants and latest CDP API

### PR DESCRIPTION
**Pull request by ambrauer**
## Description / Motivation
* Updates to reflect latest CDP API specs
* Switch from using "segments" (aka "Streaming Segments") to "variants" (aka "Experience Targeting") terminology
* Also exclude Sitecore API routes in personalize-middleware

## Testing Details
Tested locally using mocked CDP API

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts [hunk 1]
```diff
@@ -135,30 +155,33 @@ export class PersonalizeMiddleware {
...
+      personalizeInfo.contentId,
+      context,
+      browserId
+    );
     // If a browserId was not passed in (new session), a new browserId will be returned
-    browserId = segmentData.browserId;
-    // This may change, but for now we are only expected to use the first segment if there are multiple
-    segment = segmentData.segments.length ? segmentData.segments[0] : undefined;
+    browserId = variantData.browserId;
+    variantId = variantData.variantId;
```
Comment (illiakovalenko) [inline packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts:171]: Do we need to reassign `variantId`? I thought that `browserId` only can be generated

File: packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts [hunk 2]
```diff
@@ -81,10 +82,31 @@ export class PersonalizeMiddleware {
     }
   }
 
+  protected getExperienceContext(req: NextRequest): ExperienceContext {
```
Comment (illiakovalenko) [inline packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts:85]: What do you think about moving that to the `personalize/cdp-service`? It's more related to the internal functionality of the cdp-service.
so `this.cdpService.executeExperience()` can accept
```js
const variantData = await this.cdpService.executeExperience(
      personalizeInfo.contentId,
      req,
      browserId
);
```
and there you will build `context` variable

And you will not log this variable twice using `debug`

**Review by illiakovalenko — CHANGES_REQUESTED** [review]
Everything is looking good! 👍 
See few suggestions below

File: packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts (hunk 1 above)
Comment (ambrauer) [reply packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts]: No, we don't really need this variable declared early on. I've removed it and use the results from `executeExperience` call directly.

File: packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts (hunk 2 above)
Comment (ambrauer) [reply packages/sitecore-jss-nextjs/src/edge/personalize-middleware.ts:85]: The `cdpService.executeExperience` is in `sitecore-jss` package so needs to remain agnostic as to the middleware implementation making the request (imagine us adding support for Personalize to other frameworks in the future). 

However, I totally agree there was too much duplicate logging going on. I've toned it down in the latest commit.

**Review by illiakovalenko — APPROVED** [review]

**Pull request merged**